File: Android/App/app/src/main/java/br/ufscar/dc/mobile/app/model/Course.kt
package br.ufscar.dc.mobile.app.model

data class Course (
    val id: Long,
    val title: String,
    val instructor: User,
    val description: String,
    val rating: Int,
    val category: Category,
    val schedule: Schedule,
    val explanationUrl: String,
)

File: Flutter/mobile_trabalho/lib/db/entity/course.dart


// resolver problema disso ai https://floor.codes/entities/
import 'package:mobile_trabalho/db/entity/category.dart';
import 'package:mobile_trabalho/db/entity/user.dart';


class Course {
 
  String id;

  String title;
  String description;
  int rating;
  String schedule;
  String videoUrl;
  String thumbnail;
  final Category category;
  final User instructor;

  Course(
    this.id, 
    this.title, 
    this.description, 
    this.rating, 
    this.schedule,
    this.videoUrl, 
    this.thumbnail, 
    this.category, 
    this.instructor
  );
}
